packages/near-api-js/src/light-client.ts

```diff
@@ -0,0 +1,532 @@
+        if (bp.validator_stake_struct_version) {
+            const version = parseInt(
+                bp.validator_stake_struct_version.slice(1)
+            );
+            if (version !== 1) {
+                throw new Error(
+                    "Only version 1 of the validator stake struct is supported"
+                );
+            }
+        }
+        return new BorshValidatorStakeView({
+            v1: new BorshValidatorStakeViewV1({
+                account_id: bp.account_id,
+                public_key: PublicKey.fromString(bp.public_key),
+                stake: bp.stake,
+            }),
+        });
+    });
+    const serializedBps = serialize(
+        SCHEMA,
+        // NOTE: just wrapping because borsh-js requires this type to be in the schema for some reason
+        new BorshValidatorStakeViewWrapper({ bps: borshBps })
+    );
+    return crypto.createHash("sha256").update(serializedBps).digest();
+}
+
+function combineHash(h1: Uint8Array, h2: Uint8Array): Buffer {
+    const hash = crypto.createHash("sha256");
+    hash.update(h1);
+    hash.update(h2);
+    return hash.digest();
+}
+
+/**
+ * Computes the block hash given a `LightClientBlockLiteView`. Unlike the regular block header,
+ * the hash has to be calculated since it is not included in the response.
+ *
+ * @param block The block to compute the hash for.
+ */
+export function computeBlockHash(block: LightClientBlockLiteView): Buffer {
+    const header = block.inner_lite;
+    const borshHeader = new BorshBlockHeaderInnerLite({
+        height: new BN(header.height),
+        epoch_id: bs58.decode(header.epoch_id),
+        next_epoch_id: bs58.decode(header.next_epoch_id),
+        prev_state_root: bs58.decode(header.prev_state_root),
+        outcome_root: bs58.decode(header.outcome_root),
+        timestamp: new BN(header.timestamp_nanosec),
+        next_bp_hash: bs58.decode(header.next_bp_hash),
+        block_merkle_root: bs58.decode(header.block_merkle_root),
+    });
+    const msg = serialize(SCHEMA, borshHeader);
+    const innerRestHash = bs58.decode(block.inner_rest_hash);
+    const prevHash = bs58.decode(block.prev_block_hash);
+    const innerLiteHash = crypto.createHash("sha256").update(msg).digest();
+    const innerHash = combineHash(innerLiteHash, innerRestHash);
+    const finalHash = combineHash(innerHash, prevHash);
+
+    return finalHash;
+}
+
+/**
+ * Validates a light client block response from the RPC against the last known block and block
+ * producer set.
+ *
+ * @param lastKnownBlock The last light client block retrieved. This must be the block at the epoch before newBlock.
+ * @param currentBlockProducers The block producer set for the epoch of the last known block.
+ * @param newBlock The new block to validate.
+ */
+export function validateLightClientBlock(
+    lastKnownBlock: LightClientBlockLiteView,
+    currentBlockProducers: ValidatorStakeView[],
+    newBlock: NextLightClientBlockResponse
```

Comment:
For methods with multiple arguments please consider using named parameters to [avoid coupling to positional arguments](https://connascence.io/position.html), e.g.
```ts
validateLightClientBlock({ lastKnownBlock, currentBlockProducers, newBlock }: LightClientValidationOptions)
```

Reply:
I was trying to avoid creating a new type for this, but sounds good, will do!